src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala

```diff
@@ -423,4 +433,46 @@ final case class EServicesApiServiceImpl(
       }
     }
   }
+
+  private def getDocumentContentType(document: CatalogManagementDependency.EServiceDoc): Future[ContentType] =
+    ContentType
+      .parse(document.contentType)
+      .fold(
+        ex => Future.failed(ContentTypeParsingError(document.contentType, document.path, ex.map(_.formatPretty))),
+        Future.successful
+      )
+
+  private def convertToMessageEntity(documentDetails: DocumentDetails): MessageEntity = {
+    val randomPath: Path               = Files.createTempDirectory(s"document")
+    val temporaryFilePath: String      = s"${randomPath.toString}/${documentDetails.name}"
+    val file: File                     = new File(temporaryFilePath)
+    val outputStream: FileOutputStream = new FileOutputStream(file)
+    documentDetails.data.writeTo(outputStream)
+    HttpEntity.fromFile(documentDetails.contentType, file)
+  }
+
+  override def getEServiceDocumentById(eServiceId: String, descriptorId: String, documentId: String)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem],
+    toEntityMarshallerFile: ToEntityMarshaller[File]
+  ): Route = {
+    authorize(ADMIN_ROLE, API_ROLE, SECURITY_ROLE, M2M_ROLE) {
+      val operationLabel =
+        s"Retrieving EService document $documentId for EService $eServiceId and descriptor $descriptorId"
+      logger.info(operationLabel)
+
+      val result: Future[DocumentDetails] = for {
+        document    <- catalogManagementService.getEServiceDocument(eServiceId, descriptorId, documentId)
+        contentType <- getDocumentContentType(document)
```

Comment:
@beetlecrunch is this check actually necessary for a GET?

Reply:
It is used to populate the content-type header of the response. I don't think checks nothing (as far as it can detect a Content-Type).

Reply:
Also, since this check is performed by akka only I don't think it will populate it with something as specific as `application/pdf` but just `application/octet-stream`

Reply:
you're right 👍